linode/manager
# feat: [M3-6646] -  Strip whitespace on blur for username and email textfields

## Description 📝
This change comes from a reporting of the "SOA email is not valid" validation error message on the Create Domains page (https://cloud.linode.com/domains/create) if a blank space is left in the email section after an inputted email.

The cause is that we don't currently strip extra whitespace on `TextFields`. For all email and username inputs, leading and trailing whitespace is stripped when a user focuses away from the input field.

## Major Changes 🔄
- Leading and trailing whitespace is stripped from user input in the following places:
   - Domains:
      - Create Domain `email` field (http://localhost:3000/domains/create)
      - Edit Domain drawer `email` field (http://http://localhost:3000/domains)
      - Edit SOA Record drawer `email` field (http://localhost:3000/domains/{id})
   - Account:
      - Add a User drawer `username` and `email` fields (http://localhost:3000/account/users)
      - User Profile `username` and `email` fields (http://localhost:3000/account/users/{username})
      - Display Settings `username` and `email` fields (http://localhost:3000/profile/display)
 - Managed: 
   - Add and Edit Credentials drawer `username` field for Managed accounts (http://localhost:3000/managed/credentials)
   - Add and Edit Contacts drawer `email` field for Managed accounts (http://localhost:3000/managed/contacts)
   - Note: we recently added trimming for SSH keys (M3-6473), and that functionality should still work in this PR. (http://localhost:3000/profile/keys)
- Adds a boolean`trimmed` prop to the `SingleTextFieldForm` and `TextField` components, which is used in the blur event handler to trim whitespace.


## Preview 📷

| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/114685994/59e19694-1547-4cbc-bce9-0e20a6aa4967"/> | <video src="https://github.com/linode/manager/assets/114685994/e4072a68-6aef-49a9-b0f2-cda25ee507bf"/> |

## How to test 🧪
1. **How to setup test environment?**
2. **How to reproduce the issue (if applicable)?**
   - Test any/all of the textfields mentioned above and note the error messages displayed when the user enters input with a leading or trailing spaces and then focuses away from the field.
4. **How to verify changes?**
   - Test the username, email, and SSH key fields throughout the app -- see above for a list. When entering input with leading or trailing whitespaces and focusing away from the field, no error message should display. When the form is submitted, the action should complete successfully.
   - Use the MSW for Managed.
5. **How to run unit or e2e tests?**
``` 
yarn test TextField
```
